From 56fa60bec574148ae9e759bf3c4f2e4073edc0b4 Mon Sep 17 00:00:00 2001
From: James Westman <james@jwestman.net>
Date: Sat, 31 May 2025 18:56:32 -0500
Subject: [PATCH] placePopover: Fix sizing issues

Fix several sizing issues in the search popover. Removed the hard-coded
height restrictions, so it will no longer be capped at 400px if there is
enough screen space, nor will it be larger than the content requires.
This allows it to appear on small screens correctly.

Also fixed a size allocation bug. GTK requires that a popover's parent
calls present() in its size_allocate function.
---
 data/ui/place-popover.ui | 16 ++++++++++------
 src/placeEntry.js        |  5 +++++
 2 files changed, 15 insertions(+), 6 deletions(-)

diff --git a/data/ui/place-popover.ui b/data/ui/place-popover.ui
index 1e670e74..2546834a 100644
--- a/data/ui/place-popover.ui
+++ b/data/ui/place-popover.ui
@@ -13,6 +13,8 @@
     <property name="child">
       <object class="GtkStack" id="stack">
         <property name="transition-type">crossfade</property>
+        <property name="vhomogeneous">False</property>
+        <property name="interpolate-size">True</property>
         <style>
           <class name="maps-stack"/>
         </style>
@@ -39,7 +41,6 @@
               <object class="GtkListBox" id="poiMainCategoriesListBox">
                 <property name="vexpand">True</property>
                 <property name="hexpand">True</property>
-                <property name="height-request">400</property>
                 <layout>
                   <property name="row">1</property>
                   <property name="column">0</property>
@@ -58,12 +59,15 @@
           <object class="AdwSpinner" id="spinner">
             <property name="hexpand">True</property>
             <property name="vexpand">True</property>
+            <property name="width-request">48</property>
+            <property name="height-request">48</property>
+            <property name="margin-top">18</property>
+            <property name="margin-bottom">18</property>
           </object>
         </child>
         <child>
           <object class="GtkScrolledWindow" id="scrolledWindow">
             <property name="hscrollbar_policy">never</property>
-            <property name="max-content-height">400</property>
             <property name="propagate-natural-height">True</property>
             <property name="child">
               <object class="GtkListBox" id="list">
@@ -76,8 +80,8 @@
         <child>
           <object class="GtkLabel" id="noResultsLabel">
             <property name="label" translatable="yes">No results found</property>
-            <property name="width_request">16</property>
-            <property name="height_request">16</property>
+            <property name="margin-top">18</property>
+            <property name="margin-bottom">18</property>
             <style>
               <class name="dim-label"/>
             </style>
@@ -86,8 +90,8 @@
         <child>
           <object class="GtkLabel" id="errorLabel">
             <property name="label" translatable="yes">An error has occurred</property>
-            <property name="width_request">16</property>
-            <property name="height_request">16</property>
+            <property name="margin-top">18</property>
+            <property name="margin-bottom">18</property>
             <style>
               <class name="dim-label"/>
             </style>
diff --git a/src/placeEntry.js b/src/placeEntry.js
index ef89de10..a502eaaf 100644
--- a/src/placeEntry.js
+++ b/src/placeEntry.js
@@ -433,6 +433,11 @@ export class PlaceEntry extends Gtk.Entry {
         this.text = '';
         this.grab_focus();
     }
+
+    vfunc_size_allocate(width, height, baseline) {
+        super.vfunc_size_allocate(width, height, baseline);
+        this._popover.present();
+    }
 }
 
 GObject.registerClass({
-- 
GitLab

